2018-03-28/azblob/highlevel.go

```diff
@@ -487,22 +489,33 @@ func uploadStream(ctx context.Context, reader io.Reader, o UploadStreamOptions,
 			// We are at max buffers, block until we get to reuse one
 			buffer = <-incoming
 		}
-		n, err := io.ReadFull(reader, buffer)
-		if err != nil {
+		var n int
+		n, readErr = io.ReadFull(reader, buffer)
+		if readErr != nil {
+			if readErr != io.EOF && readErr != io.ErrUnexpectedEOF {
```

Comment:
why io.ErrUnexpectedEOF?

Reply:
From the docs

```
func ReadFull(r Reader, buf []byte) (n int, err error)
    ReadFull reads exactly len(buf) bytes from r into buf. It returns the number
    of bytes copied and an error if fewer bytes were read. The error is EOF only
    if no bytes were read. If an EOF happens after reading some but not all the
    bytes, ReadFull returns ErrUnexpectedEOF. On return, n == len(buf) if and
    only if err == nil.
```

Since we aren't expecting the input buffer to be a multiple of `len(buffer)` in size then `is.ErrUnexpectedEOF` is an expected successful termination as well as `io.EOF`.